cmd/generate-catalog/generate.go

```diff
@@ -379,24 +359,24 @@ func validateImageReference(imageRef string) error {
 		return fmt.Errorf("cannot parse string as container image reference %s: %w", imageRef, err)
 	}
 
-	// check that digest is provided
-	if _, ok := ref.(reference.Canonical); !ok {
+	canonical, ok := ref.(reference.Canonical)
+	if !ok || canonical.Digest() == "" {
 		return fmt.Errorf("image reference %s does not include a digest", imageRef)
 	}
-
-	named, ok := ref.(reference.Named)
-	if !ok {
-		return fmt.Errorf("image reference %s has invalid format", imageRef)
+	if ok && canonical.Digest().Algorithm() != digest.SHA256 {
```

Comment:
I believe, there's no need to check for `ok` at this point.
```suggestion
	if canonical.Digest().Algorithm() != digest.SHA256 {
```